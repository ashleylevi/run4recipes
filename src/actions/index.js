export const loadAllRecipes = (recipes1, recipes2, recipes3) => ({
  type: 'LOAD_ALL_RECIPES',
  recipes1,
  recipes2,
  recipes3
})

export const loadPastaRecipes = (recipes) => ({
  type: 'LOAD_PASTA_RECIPES',
  recipes
})

export const loadPotatoRecipes = (recipes) => ({
  type: 'LOAD_POTATO_RECIPES',
  recipes
})

export const loadBreadRecipes = (recipes) => ({
  type: 'LOAD_BREAD_RECIPES',
  recipes
})

export const loading = (bool) => ({
  type: "LOADING",
  bool
})

export const searchRecipes = (searchValue) => ({
  type: "SEARCH_RECIPES",
  searchValue
})
