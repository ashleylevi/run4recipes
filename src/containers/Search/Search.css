input {
  background-color: #D5AD47;
  padding-left: 20px;
  height: 47px;
  border: none;
  font-size: 17px;
  /* width: 99.5%; */
}

input:hover {
  cursor: pointer;
}

input::placeholder {
  color: #32636E;
}

input[type=text] {
  padding-left: 10px;
}