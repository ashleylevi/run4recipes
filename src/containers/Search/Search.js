import React, { Component } from 'react';
import { connect } from 'react-redux';
import { searchRecipes } from '../../actions/';
import './Search.css';

class Search extends Component {
  constructor() {
    super()
    this.state = {
      searchValue: ''
    }
  }

  handleChange = (e) => {
    this.setState({
      searchValue: e.target.value
    })
    this.props.searchRecipes(e.target.value.toLowerCase())
  }

  render() {
    return (
      <div className="search-div">
        <form>
          <input
          type="text"
          placeholder="Search Recipes"
          value={this.state.searchValue}
          onChange={this.handleChange}
          className="search-input">
          </input>
        </form>    
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  searchRecipes: (searchValue) => dispatch(searchRecipes(searchValue))
})

export default connect(null, mapDispatchToProps)(Search)