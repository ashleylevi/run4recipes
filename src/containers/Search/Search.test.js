import { mapDispatchToProps } from './Search';
import { Search } from './Search';
import React from 'react';
import { shallow } from 'enzyme';
import { searchRecipes } from '../../actions/index';

describe('Search Component', () => {
  let wrapper;
  let searchRecipesMock;
  
  beforeEach(() => {
    searchRecipesMock = jest.fn()
    wrapper = shallow(<Search searchRecipes={searchRecipesMock} />)
  })
  
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should match the snapshot when there is a searchValue', () => {
    wrapper.setState({searchValue: 'kimchi'})
    wrapper.update()
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    let wrapper = shallow(<Search searchRecipes={searchRecipesMock} />)
    const expected = {
      searchValue: ''
    }
    console.log(wrapper.state())
    expect(wrapper.instance().state).toEqual(expected)
  })

  describe('handleChange', () => {
    const mockEvent = {target: {value: 'kimchi'}}

    it('it should update searchValue in state', () => {
      const expected = 'kimchi'
      wrapper.instance().handleChange(mockEvent)
      expect(wrapper.state().searchValue).toEqual(expected)
    })
  })
  

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a searchRecipes action', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = searchRecipes('kimchi')
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.searchRecipes('kimchi')
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
  
}) 