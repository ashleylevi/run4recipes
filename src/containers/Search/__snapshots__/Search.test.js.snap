// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    searchRecipes={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search Recipes"
          type="text"
          value=""
        />
      </form>,
      "className": "search-div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search Recipes"
          type="text"
          value=""
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Search Recipes",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search Recipes"
            type="text"
            value=""
          />
        </form>,
        "className": "search-div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search Recipes"
            type="text"
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Search Recipes",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Search Component should match the snapshot when there is a searchValue 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    searchRecipes={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search Recipes"
          type="text"
          value="kimchi"
        />
      </form>,
      "className": "search-div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search Recipes"
          type="text"
          value="kimchi"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Search Recipes",
          "type": "text",
          "value": "kimchi",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search Recipes"
            type="text"
            value="kimchi"
          />
        </form>,
        "className": "search-div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search Recipes"
            type="text"
            value="kimchi"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Search Recipes",
            "type": "text",
            "value": "kimchi",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
